protocol/signalfx/tagreplace/tagreplace.go

```diff
@@ -0,0 +1,84 @@
+package tagreplace
+
+import (
+	"context"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/datapoint/dpsink"
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/golib/pointer"
+	"github.com/signalfx/golib/trace"
+	"regexp"
+	"strings"
+)
+
+type sink interface {
+	dpsink.Sink
+	trace.Sink
+}
+
+var _ sink = &TagReplace{}
+
+// TagReplace does regex search and replace in span Names and puts resultant in tags.
+// This modifies the objects parsed, so in a concurrent context, you will want to copy the objects sent here first
+type TagReplace struct {
+	rules []*regexp.Regexp
+	next  sink
+}
+
+// AddDatapoints is a passthrough
+func (t *TagReplace) AddDatapoints(ctx context.Context, points []*datapoint.Datapoint) error {
+	return t.next.AddDatapoints(ctx, points)
+}
+
+// AddEvents is a passthrough
+func (t *TagReplace) AddEvents(ctx context.Context, events []*event.Event) error {
+	return t.next.AddEvents(ctx, events)
+}
+
+// AddSpans maps all rules to all spans and does the replacements, this can be VERY expensive, and modifies the spans
+func (t *TagReplace) AddSpans(ctx context.Context, spans []*trace.Span) error {
+	for _, s := range spans {
+		if s.Name != nil {
+			for _, r := range t.rules {
+				nms := r.SubexpNames()
+				oldName := *s.Name
+				if ms := r.FindStringSubmatch(oldName); ms != nil {
+					ims := r.FindStringSubmatchIndex(oldName)
+					newTags := map[string]string{}
+
+					var newName []string
+					var index = 0
+
+					for i := 1; i < len(ms); i++ {
+						newTags[nms[i]] = ms[i]
```

Comment:
Any particular reason why you can't modify `s.Tags` directly?

Reply:
Yeah, that's fine.